testfixtures/logcapture.py

```diff
@@ -73,11 +86,53 @@ def atexit(cls):
                 '%s' % ('\n'.join((str(i.names) for i in cls.instances)))
                 )
 
+    def __len__(self):
+        return len(self.records)
+
+    def __getitem__(self, key):
+        return self._rec_2_obj(self.records[key])
+
+    def __contains__(self, what):
+        for i, item in enumerate(self):
+            if what == item:
+                self.records[i].is_asserted = True
+                return True
+        return False
+
     def clear(self):
         """Clear any entries that have been captured."""
         self.records = []
 
+    def mark_logs_as_asserted(self):
+        """ Mark all captured logs as asserted,
+         use this when doing the assertion externally """
+        for record in self.records:
+            record.is_asserted = True
+
+    def assert_no_un_asserted_logs(self, level=None):
+        """
+        raise  AssertionError if there are un-asserted logs with level greater or equal wit level
+        :param level: the logging level, defaults to self.ensure_assertions_for
+        """
+        if level is None:
+            level = self.ensure_assertions_for_level
+        if level == logging.NOTSET:
+            return
+        un_asserted = []
+        for record in self.records:
+            if level <= record.levelno and not record.is_asserted:
+                un_asserted.append(self._rec_2_obj(record))
+        if un_asserted:
+            raise AssertionError((
+                    'Not asserted ERROR log(s): %s'
+                ) % (pformat(un_asserted)))
+
     def emit(self, record):
+        """
+        :param record: a LogRecord instance
+        :type record: logging.LogRecord
+        """
+        record.is_asserted = False
         self.records.append(record)
```

Comment:
Lose this docstring, it's not part of the public api.

Reply:
I have added these as an aid for the next dev (including myself in the future) who will look at this code. I do not see why private apis should not be documented, but if you feel strongly about it I will remove them.

Reply:
I'm afraid I do: if it's a docstring, it shows up in the API docs and forms part of the public API, which I'd like to avoid for these methods.

Comments are fine to help future developers...

Reply:
Ok, I understand now, thanks for the explanation.